### PR TITLE
feat(pubsub): Add publisher/subscriber system

### PR DESCRIPTION
**Pull request by FlorentinDUBOIS**
Add publisher system
Add subscriber system
Add tests to each

### CODE REVIEW

File: src/index.spec.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import 'reflect-metadata'
```
Comment (miton18) [inline src/index.spec.ts:1]: what are you importing ?

File: src/inversify.config.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import 'reflect-metadata'
```
Comment (miton18) [inline src/inversify.config.ts:1]: what are you importing ?

File: src/redis/impl/RedisSubscriber.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  public subscribe<T>(channel: string): Observable<T> {
+    return Observable.create((observer: Observer<T>): void => {
+      this.$redis.subscribe(channel, (error: Error, count: number): void => {
+        if (error) {
+          return observer.error(error)
+        }
+      })
+
+      this.$redis.on('message', (chan: string, message: T): void => {
+        if (channel === chan) {
```
Comment (miton18) [inline src/redis/impl/RedisSubscriber.ts:19]: to be test but useless, if you subscribe on 1 chan youget messges for the same chan

**Review by miton18 — CHANGES_REQUESTED** [review]
smell very good

File: src/redis/impl/RedisSubscriber.ts (hunk 3 above)
Comment (FlorentinDUBOIS) [reply src/redis/impl/RedisSubscriber.ts:19]: in case of

**Pull request merged**